src/components/views/rooms/RoomSettings.js

```diff
@@ -141,11 +268,96 @@ module.exports = React.createClass({
         });
     },
 
+    onAliasChanged: function(domain, index, alias) {
+        if (alias === "") return; // hit the delete button to delete please
+        var oldAlias;
+        if (this.isAliasValid(alias)) {
+            oldAlias = this.state.aliases[domain][index];
+            this.state.aliases[domain][index] = alias;
+            this.setState({ aliases_changed : true });
+        }
+        else {
+            var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");            
+            Modal.createDialog(ErrorDialog, {
+                title: "Invalid alias format", 
+                description: "'" + alias + "' is not a valid format for an alias",
+            });
+        }        
+    },
+
+    onAliasDeleted: function(domain, index) {
+        // XXX: can we edit state directly and then set, or should we copy it first?
+        var alias = this.state.aliases[domain].splice(index, 1);
+        this.setState({ 
+            aliases: this.state.aliases
+        });
+
+        this.setState({ aliases_changed : true });
+    },
+
+    onAliasAdded: function(alias) {
```

Comment:
Clients can only add aliases on their *local* HS, but it looks like this will allow them to be added to any server. This creates needless typing/remembering for the suffix part.